src/main/java/de/zalando/aruha/nakadi/partitioning/RandomPartitioningStrategy.java

```diff
@@ -0,0 +1,32 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.ExceptionWrapper;
+import de.zalando.aruha.nakadi.exceptions.InvalidPartitioningKeyFieldsException;
+import de.zalando.aruha.nakadi.exceptions.PartitioningException;
+import de.zalando.aruha.nakadi.util.JsonPathAccess;
+import org.json.JSONObject;
+
+import java.util.List;
+import java.util.Random;
+
+import static de.zalando.aruha.nakadi.exceptions.ExceptionWrapper.wrapFunction;
+import static java.lang.Math.abs;
+
+public class RandomPartitioningStrategy implements PartitioningStrategy {
+
+    private static final Random random = new Random();
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions) {
+        if (partitions.size() == 1) {
```

Comment:
That is true, the only reason why I made this branch is to save some performance here (as the random will still be calculated even that only one value is possible)
If most of event-types will have only one partition - this will be called quite often.